test/EndToEnd/ProjectBuildTests.cs

```diff
@@ -156,58 +138,242 @@ public void ItCanPublishArm64Wpf()
+        [InlineData("reactredux")]
+        public void ItCanCreateTemplateWithDefaultFramework(string templateName)
+        {
+            string framework = DetectExpectedDefaultFramework();
+            TestTemplateCreateAndBuild(templateName, build: false, framework: framework);
+        }
+
+        /// <summary>
+        /// [Windows only tests]
+        /// The test checks if the template creates the template for correct framework by default.
+        /// For .NET 6 the templates should create the projects targeting net6.0.
+        /// </summary>
+        [WindowsOnlyTheory]
+        [InlineData("wpf")]
+        [InlineData("wpf", "C#")]
+        [InlineData("wpf", "VB")]
+        [InlineData("wpflib")]
+        [InlineData("wpflib", "C#")]
+        [InlineData("wpflib", "VB")]
+        [InlineData("wpfcustomcontrollib")]
+        [InlineData("wpfcustomcontrollib", "C#")]
+        [InlineData("wpfcustomcontrollib", "VB")]
+        [InlineData("wpfusercontrollib")]
+        [InlineData("wpfusercontrollib", "C#")]
+        [InlineData("wpfusercontrollib", "VB")]
+        [InlineData("winforms")]
+        [InlineData("winforms", "C#")]
+        [InlineData("winforms", "VB")]
+        [InlineData("winformslib")]
+        [InlineData("winformslib", "C#")]
+        [InlineData("winformslib", "VB")]
+        [InlineData("winformscontrollib")]
+        [InlineData("winformscontrollib", "C#")]
+        [InlineData("winformscontrollib", "VB")]
+        public void ItCanCreateAndBuildTemplatesWithDefaultFramework_Windows(string templateName, string language = "")
+        {
+            string framework = DetectExpectedDefaultFramework();
+            TestTemplateCreateAndBuild(templateName, selfContained: true, language: language, framework: $"{framework}-windows");
+        }
+
+        /// <summary>
+        /// [project is not built on linux-musl]
+        /// The test checks if the template creates the template for correct framework by default.
+        /// For .NET 6 the templates should create the projects targeting net6.0.
+        /// </summary>
+        [Theory]
+        [InlineData("grpc")]
+        public void ItCanCreateAndBuildTemplatesWithDefaultFramework_DisableBuildOnLinuxMusl(string templateName)
+        {
+            string framework = DetectExpectedDefaultFramework();
+
+            if (RuntimeInformation.RuntimeIdentifier.StartsWith("alpine")) //linux musl
+            {
+                TestTemplateCreateAndBuild(templateName, build: false, framework: framework);
+            }
+            else
+            {
+                TestTemplateCreateAndBuild(templateName, selfContained: true, framework: framework);
+            }
+        }
+
+        private static string DetectExpectedDefaultFramework()
+        {
+            string dotnetFolder = Path.GetDirectoryName(RepoDirectoriesProvider.DotnetUnderTest);
+            string[] sdkFolders = Directory.GetDirectories(Path.Combine(dotnetFolder, "sdk"));
```

Comment:
I'd suggest looking at the latest version folder under `shared/Microsoft.NETCore.App` instead of looking at the SDK version.